One more formatting problem left:
```
Formatting [652/1630] librz/util/sdb/src/cdb_make.c
librz/util/sdb/src/cdb_make.c:13:18: error: code should be clang-formatted [-Wclang-format-violations]
#define ALIGNMENT sizeof(void *)
                 ^
```
Just use `sys/clang-format.py`

--- a/librz/include/rz_endian.h
+++ b/librz/include/rz_endian.h
@@ -259,6 +259,11 @@ static inline void rz_write_le24(void *_dest, ut32 val) {
 	rz_write_le8(dest, val >> 16);
 }
 
+/**
+ * \brief Unpack the buffer, in low-endian order, into a 32-bit integer.
+ * \param[in] src The buffer to read from.
+ * \return The result of unpacking \p src.
+ */
 static inline ut32 rz_read_le32(const void *src) {
 	if (!src) {
 		return UT32_MAX;
@@ -276,6 +281,11 @@ static inline ut32 rz_read_at_le32(const void *src, size_t offset) {
 	return rz_read_le32(s);
 }
 
+/**
+ * \brief Pack the 32-bit integer in little-endian order into a buffer.
+ * \param[out] dest The buffer to write to.
+ * \param val The integer to pack.
+ */
 static inline void rz_write_le32(void *dest, ut32 val) {
 	rz_write_le16(dest, val);
 	rz_write_at_le16(dest, val >> 16, sizeof(ut16));

--- a/librz/util/sdb/src/cdb.c
+++ b/librz/util/sdb/src/cdb.c
@@ -2,6 +2,17 @@
 // SPDX-FileCopyrightText: 2014-2016 pancake <pancake@nopcode.org>
 // SPDX-License-Identifier: CC-PDDC
 
+/**
+ * \internal
+ * \file
+ * \brief Reading and operating on a \ref cdb structure.
+ * \attention This API is not reentrant.
+ *
+ * The \ref cdb structure is an associative array mapping strings to
+ * strings. Originally written by D. J. Bernstein, see
+ * <https://cr.yp.to/cdb.html> for a description of the binary format.
+ */
+
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -11,7 +22,22 @@
 #endif
 #include "sdb_private.h"
 
-/* XXX: this code must be rewritten . too slow */
+/**
+ * \brief Read the key and value lengths at a given offset.
+ * \param c The \def cdb structure.
+ * \param[out] klen The 8-bit key length read.
+ * \param[out] vlen The 24-bit value length read.
+ * \param pos The offset in the \ref cdb structure to read from.
+ * \return True if successfully read.
+ *
+ * The 8-bit and 24-bit integer are read in sequence. The 24-bit
+ * integer is expected to be stored in low endian order.
+ *
+ * Thus in total, four bytes are read, and ahead these four bytes,
+ * there is expected to be a key-value pair (which is not read). The
+ * first 8-bit integer read is the length of the key, and the 24-bit
+ * integer read after is the length of the value that follows the key.
+ */
 bool cdb_getkvlen(struct cdb *c, ut32 *klen, ut32 *vlen, ut32 pos) {
 	ut8 buf[4] = { 0 };
 	*klen = *vlen = 0;
@@ -27,6 +53,29 @@ bool cdb_getkvlen(struct cdb *c, ut32 *klen, ut32 *vlen, ut32 pos) {
 	return true;
 }
 
+/**
+ * \brief Deallocate a \ref cdb structure.
+ * \param c The \ref cdb structure to deallocate.
+ *
+ * Deallocates the memory used by the structure. This function may be
+ * called on any \ref cdb structure that has been previously
+ * initialized with a call to \ref cdb_init.
+ *
+ * \attention Any initialized \cdb structure should eventually be
+ * freed according to this function in order to release its memory.
+ *
+ * # Example
+ *
+ * \code{.c}
+ * // fd is a file descriptor from some open file.
+ * struct cdb c;
+ * c.fd = -1;
+ * if(cdb_init(&c, fd)) {
+ *   // ... use c as desired
+ *   cdb_free(&c);
+ * }
+ * \endcode
+ */
 void cdb_free(struct cdb *c) {
 	if (!c->map) {
 		return;
@@ -39,6 +88,10 @@ void cdb_free(struct cdb *c) {
 	c->map = NULL;
 }
 
+/**
+ * \brief Reset the search state of the \ref cdb structure to its beginning.
+ * \param c The \ref cdb structure operated upon.
+ */
 void cdb_findstart(struct cdb *c) {
 	c->loop = 0;
 #if !HAVE_HEADER_SYS_MMAN_H
@@ -48,6 +101,43 @@ void cdb_findstart(struct cdb *c) {
 #endif
 }
 
+/**
+ * \brief Initialize a \ref cdb structure.
+ * \param c A pointer to the \ref cdb structure to allocate.
+ * \param fd The file descriptor containing the database.
+ * \return \c true if initialization succeeded, \c false otherwise.
+ *
+ * This is the first operation on a \ref cdb structure. It should be
+ * called on a \ref cdb structure where \ref cdb.fd is set to \c -1.
+ *
+ * This function may be called more than once on the same
+ * structure. It will close the associated file descriptor and
+ * deallocate memory before associating with the new file descriptor
+ * and allocating new memory.
+ *
+ * \attention The \p fd parameter should be kept open at least until
+ * the \ref cdb structure is disassociated with it.
+ * \attention The function \ref cdb_free should be called once the
+ * \cdb structure is no longer needed.
+ *
+ * # Example
+ * \code{.c}
+ * int fd;
+ * struct cdb c;
+ * c.fd = -1; // This is important in order to call cdb_init.
+ * fd = open("cdb_file", O_RDONLY);
+ * if(cdb_init(&c, fd)) {
+ *   // The cdb structure is ready to use.
+ *   // ...
+ * } else {
+ *   // report error
+ * }
+ * close(fd);
+ * // From this point below, ensure that the cdb structure is either
+ * // reinitialized with cdb_init or deallocated with cdb_dealloc.
+ * // Do not call other operations once the file descriptor is closed!
+ * \endcode
+ */
 bool cdb_init(struct cdb *c, int fd) {
 	struct stat st;
 	if (fd != c->fd && c->fd != -1) {
@@ -86,6 +176,18 @@ bool cdb_init(struct cdb *c, int fd) {
 	return false;
 }
 
+/**
+ * \brief Read bytes from the \ref cdb structure.
+ * \param c The \def cdb structure.
+ * \param[out] buf The buffer to write to.
+ * \param len Number of bytes to write to \p buf.
+ * \param pos The offset in the \ref cdb structure to read from.
+ * \return \c true if \p bytes were successfully read, \c false
+ *  otherwise.
+ *
+ * The function writes \p len bytes to \p buf from the offset \p in
+ * the \ref cdb structure.
+ */
 bool cdb_read(struct cdb *c, char *buf, ut32 len, ut32 pos) {
 	if (c->map) {
 		if ((pos > c->size) || (c->size - pos < len)) {
@@ -111,6 +213,12 @@ bool cdb_read(struct cdb *c, char *buf, ut32 len, ut32 pos) {
 	return true;
 }
 
+/**
+ * \brief The match function to compare keys.
+ * \return The value \c 1 is returned if the keys match byte-for-byte.
+ * \return The value \c 0 is returned if the keys do not match.
+ * \return The value \c -1 is returned if there was an error.
+ */
 static int match(struct cdb *c, const char *key, ut32 len, ut32 pos) {
 	char buf[32];
 	const size_t szb = sizeof buf;
@@ -129,16 +237,45 @@ static int match(struct cdb *c, const char *key, ut32 len, ut32 pos) {
 	return 1;
 }
 
+/**
+ * \brief Find the next occurence of \p key in the \ref cdb structure.
+ * \param u The hash of the \p key to search for.
+ * \param key The key, a NUL-terminated string.
+ * \param len The string length of \p key.
+ * \return The function returns 1 when the key is matched, 0 when not
+ * found and -1 on error.
+ *
+ * This function may be used repeatedly until no more matches are
+ * produced. The state of the search is saved within the \ref cdb
+ * structure, and may be reset with \ref cdb_findstart.
+ *
+ * # Example
+ * \code{.c}
+ * const char *key = "MYKEY";
+ * ut32 len = (ut32)strlen(key);
+ * ut32 hash = my_hash_func(key);
+ * int r;
+ * cdb_findstart(&c);
+ * while((r = cdb_findnext(&c, hash, key, len)) == 1) {
+ *   // process match; details in c->dpos, c->dlen
+ * }
+ * if(r == -1) {
+ *   // process error
+ * }
+ * cdb_findstart(&c); // reset the search state if necessary
+ * \endcode
+ */
 int cdb_findnext(struct cdb *c, ut32 u, const char *key, ut32 len) {
 	char buf[8];
 	ut32 pos;
 	int m;
-	len++;
+	len++; // To include the \0 byte.
 	if (c->fd == -1) {
 		return -1;
 	}
 	c->hslots = 0;
 	if (!c->loop) {
+		/* Read the hash table position. */
 		const int bufsz = ((u + 1) & 0xFF) ? sizeof(buf) : sizeof(buf) / 2;
 		if (!cdb_read(c, buf, bufsz, (u << 2) & 1023)) {
 			return -1;
@@ -162,6 +299,7 @@ int cdb_findnext(struct cdb *c, ut32 u, const char *key, ut32 len) {
 		c->kpos = c->hpos + u;
 	}
 	while (c->loop < c->hslots) {
+		/* Loop through the hash table slots. */
 		if (!cdb_read(c, buf, sizeof(buf), c->kpos)) {
 			return 0;
 		}
@@ -176,6 +314,7 @@ int cdb_findnext(struct cdb *c, ut32 u, const char *key, ut32 len) {
 		}
 		ut32_unpack(buf, &u);
 		if (u == c->khash) {
+			/* The hashes match, compare the strings. */
 			if (!cdb_getkvlen(c, &u, &c->dlen, pos) || !u) {
 				return -1;
 			}

--- a/librz/util/sdb/src/cdb.c
+++ b/librz/util/sdb/src/cdb.c
@@ -16,6 +16,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <rz_endian.h>
 #include "cdb.h"
 #if HAVE_HEADER_SYS_MMAN_H
 #include <sys/mman.h>
@@ -281,9 +282,9 @@ int cdb_findnext(struct cdb *c, ut32 u, const char *key, ut32 len) {
 			return -1;
 		}
 		/* hslots = (hpos_next - hpos) / 8 */
-		ut32_unpack(buf, &c->hpos);
+		c->hpos = rz_read_le32(buf);
 		if (bufsz == sizeof(buf)) {
-			ut32_unpack(buf + 4, &pos);
+			pos = rz_read_at_le32(buf, 4);
 		} else {
 			pos = c->size;
 		}
@@ -303,7 +304,7 @@ int cdb_findnext(struct cdb *c, ut32 u, const char *key, ut32 len) {
 		if (!cdb_read(c, buf, sizeof(buf), c->kpos)) {
 			return 0;
 		}
-		ut32_unpack(buf + 4, &pos);
+		pos = rz_read_at_le32(buf, 4);
 		if (!pos) {
 			return 0;
 		}
@@ -312,7 +313,7 @@ int cdb_findnext(struct cdb *c, ut32 u, const char *key, ut32 len) {
 		if (c->kpos == c->hpos + (c->hslots << 3)) {
 			c->kpos = c->hpos;
 		}
-		ut32_unpack(buf, &u);
+		u = rz_read_le32(buf);
 		if (u == c->khash) {
 			/* The hashes match, compare the strings. */
 			if (!cdb_getkvlen(c, &u, &c->dlen, pos) || !u) {

--- a/librz/util/sdb/src/cdb.h
+++ b/librz/util/sdb/src/cdb.h
@@ -1,6 +1,16 @@
 // SPDX-FileCopyrightText: D. J. Bernstein <djb@cr.yp.to>
 // SPDX-License-Identifier: CC-PDDC
 
+/**
+ * \internal
+ * \file
+ * \brief Reading and operating on a \ref cdb structure.
+ *
+ * The \ref cdb structure is an associative array mapping strings to
+ * strings. Originally written by D. J. Bernstein, see
+ * <https://cr.yp.to/cdb.html> for a description of the binary format.
+ */
+
 #ifndef CDB_H
 #define CDB_H
 
@@ -11,26 +21,37 @@
 extern "C" {
 #endif
 
+/**
+ * \def KVLSZ
+ * \brief The size in bytes of the key-value lengths combined.
+ */
+
 #define KVLSZ         4
 #define CDB_MAX_KEY   0xff
 #define CDB_MAX_VALUE 0xffffff
 
 #define CDB_HASHSTART 5381
 
+/** \internal
+ * \brief The \ref cdb structure.
+ *
+ * An associative array of strings to strings based on cdb by
+ * D. J. Bernstein, see <https://cr.yp.to/cdb.html>.
+ */
 struct cdb {
-	char *map; /* 0 if no map is available */
-	int fd; /* filedescriptor */
-	ut32 size; /* initialized if map is nonzero */
-	ut32 loop; /* number of hash slots searched under this key */
-	ut32 khash; /* initialized if loop is nonzero */
-	ut32 kpos; /* initialized if loop is nonzero */
-	ut32 hpos; /* initialized if loop is nonzero */
-	ut32 hslots; /* initialized if loop is nonzero */
-	ut32 dpos; /* initialized if cdb_findnext() returns 1 */
-	ut32 dlen; /* initialized if cdb_findnext() returns 1 */
+	char *map; ///< Maps the file in memory. NULL if no map is available.
+	int fd; ///< The file descriptor from which the cdb structure is read.
+	ut32 size; ///< Initialized if map is nonzero.
+	ut32 loop; ///< The search state, number of hash slots searched in a given key.
+	ut32 khash; ///< Hash of key. Initialized if loop is nonzero.
+	ut32 kpos; ///< Key position. Initialized if loop is nonzero.
+	ut32 hpos; ///< Current hash table position. Initialized if loop is nonzero.
+	ut32 hslots; ///< Number of slots of current hash table. Initialized if loop is nonzero.
+	ut32 dpos; ///< Data position. Initialized if cdb_findnext() returns 1.
+	ut32 dlen; ///< Data length. Initialized if cdb_findnext() returns 1.
 };
 
-/* TODO THIS MUST GTFO! */
+/* TODO Remove this! */
 bool cdb_getkvlen(struct cdb *db, ut32 *klen, ut32 *vlen, ut32 pos);
 void cdb_free(struct cdb *);
 bool cdb_init(struct cdb *, int fd);

--- a/librz/util/sdb/src/cdb_make.c
+++ b/librz/util/sdb/src/cdb_make.c
@@ -3,6 +3,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <rz_endian.h>
 #include "sdb.h"
 #include "cdb.h"
 #include "cdb_make.h"
@@ -153,7 +154,7 @@ int cdb_make_finish(struct cdb_make *c) {
 	for (i = 0; i < 256; i++) {
 		count = c->count[i];
 		len = count << 1;
-		ut32_pack(c->final + 4 * i, c->pos);
+		rz_write_at_le32(c->final, c->pos, 4 * i);
 		for (u = 0; u < len; u++) {
 			c->hash[u].h = c->hash[u].p = 0;
 		}
@@ -168,8 +169,8 @@ int cdb_make_finish(struct cdb_make *c) {
 			c->hash[where] = *hp++;
 		}
 		for (u = 0; u < len; u++) {
-			ut32_pack(buf, c->hash[u].h);
-			ut32_pack(buf + 4, c->hash[u].p);
+			rz_write_le32(buf, c->hash[u].h);
+			rz_write_at_le32(buf, c->hash[u].p, 4);
 			if (!buffer_putalign(&c->b, buf, 8)) {
 				return 0;
 			}

--- a/librz/util/sdb/src/cdb_make.c
+++ b/librz/util/sdb/src/cdb_make.c
@@ -4,36 +4,14 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <rz_endian.h>
+#include <rz_util/rz_alloc.h>
 #include "sdb.h"
 #include "cdb.h"
 #include "cdb_make.h"
 #include "sdb_private.h"
 
-#if __APPLE__
-// for MAC_OS_X_VERSION_10_6
-#include <AvailabilityMacros.h>
-#endif
-
-#define ALIGNMENT sizeof(void *)
-
-char *cdb_alloc(ut32 n) {
-#if __APPLE__ && defined(MAC_OS_X_VERSION_10_6)
-	void *ret = NULL;
-	return posix_memalign(&ret, ALIGNMENT, n) ? NULL : ret;
-#elif __WINDOWS__ && !__CYGWIN__
-	return _aligned_malloc(n, ALIGNMENT);
-#else
-	return malloc(n);
-#endif
-}
-
-void cdb_alloc_free(void *x) {
-#if __WINDOWS__ && !__CYGWIN__
-	_aligned_free(x);
-#else
-	free(x);
-#endif
-}
+#define ALIGNMENT     sizeof(void *)
+#define RZ_ANEW(x, n) (x *)rz_malloc_aligned(n * sizeof(x), ALIGNMENT)
 
 int cdb_make_start(struct cdb_make *c, int fd) {
 	int i;
@@ -60,12 +38,11 @@ static inline int incpos(struct cdb_make *c, ut32 len) {
 	return 1;
 }
 
-#define R_ANEW(x) (x *)cdb_alloc(sizeof(x))
 int cdb_make_addend(struct cdb_make *c, ut32 keylen, ut32 datalen, ut32 h) {
 	ut32 u;
 	struct cdb_hplist *head = c->head;
 	if (!head || (head->num >= CDB_HPLIST)) {
-		if (!(head = R_ANEW(struct cdb_hplist))) {
+		if (!(head = RZ_ANEW(struct cdb_hplist, 1))) {
 			return 0;
 		}
 		head->num = 0;
@@ -133,7 +110,7 @@ int cdb_make_finish(struct cdb_make *c) {
 	if (memsize > (UT32_MAX / sizeof(struct cdb_hp))) {
 		return 0;
 	}
-	c->split = (struct cdb_hp *)cdb_alloc(memsize * sizeof(struct cdb_hp));
+	c->split = (struct cdb_hp *)RZ_ANEW(struct cdb_hp, memsize);
 	if (!c->split) {
 		return 0;
 	}
@@ -186,12 +163,12 @@ int cdb_make_finish(struct cdb_make *c) {
 	if (!seek_set(c->fd, 0)) {
 		return 0;
 	}
-	// free childs
+	// free children
 	for (x = c->head; x;) {
 		n = x->next;
-		cdb_alloc_free(x);
+		rz_free_aligned(x);
 		x = n;
 	}
-	cdb_alloc_free(c->split);
+	rz_free_aligned(c->split);
 	return buffer_putflush(&c->b, c->final, sizeof c->final);
 }

--- a/librz/util/sdb/src/sdb_private.h
+++ b/librz/util/sdb/src/sdb_private.h
@@ -1,6 +1,12 @@
 // SPDX-FileCopyrightText: pancake <pancake@nopcode.org>
 // SPDX-License-Identifier: MIT
 
+/**
+ * \internal
+ * \file
+ * \brief Some helpful macros and static functions.
+ */
+
 #ifndef SDB_PRIVATE_H_
 #define SDB_PRIVATE_H_
 
@@ -22,10 +28,21 @@ extern "C" {
 #define write_(fd, buf, count) SDB_V_NOT(write(fd, buf, count), -1)
 #define read_(fd, buf, count)  SDB_V_NOT(read(fd, buf, count), -1)
 
+/**
+ * \brief Reposition file offset.
+ * \param fd The file whose offset is to be repositioned
+ * \param pos The offset to reposition the file to.
+ * \return 1 if successful, 0 otherwise.
+ */
 static inline int seek_set(int fd, off_t pos) {
 	return ((fd == -1) || (lseek(fd, (off_t)pos, SEEK_SET) == -1)) ? 0 : 1;
 }
 
+/**
+ * \brief Pack the 32-bit integer in little-endian order into a buffer.
+ * \param[out] s The buffer to write to.
+ * \param u The integer to pack.
+ */
 static inline void ut32_pack(char s[4], ut32 u) {
 	s[0] = u & 255;
 	u >>= 8;
@@ -35,6 +52,11 @@ static inline void ut32_pack(char s[4], ut32 u) {
 	s[3] = u >> 8;
 }
 
+/**
+ * \brief Pack the 32-bit integer in big-endian order into a buffer.
+ * \param[out] s The buffer to write to.
+ * \param u The integer to pack.
+ */
 static inline void ut32_pack_big(char s[4], ut32 u) {
 	s[3] = u & 255;
 	u >>= 8;
@@ -44,6 +66,11 @@ static inline void ut32_pack_big(char s[4], ut32 u) {
 	s[0] = u >> 8;
 }
 
+/**
+ * \brief Unpack the buffer, in low-endian order, into a 32-bit integer.
+ * \param[in] s The buffer to read from.
+ * \param[out] u The result of unpacking \p s.
+ */
 static inline void ut32_unpack(char s[4], ut32 *u) {
 	ut32 result = 0;
 	result = (ut8)s[3];

--- a/librz/util/sdb/src/sdb_private.h
+++ b/librz/util/sdb/src/sdb_private.h
@@ -38,51 +38,6 @@ static inline int seek_set(int fd, off_t pos) {
 	return ((fd == -1) || (lseek(fd, (off_t)pos, SEEK_SET) == -1)) ? 0 : 1;
 }
 
-/**
- * \brief Pack the 32-bit integer in little-endian order into a buffer.
- * \param[out] s The buffer to write to.
- * \param u The integer to pack.
- */
-static inline void ut32_pack(char s[4], ut32 u) {
-	s[0] = u & 255;
-	u >>= 8;
-	s[1] = u & 255;
-	u >>= 8;
-	s[2] = u & 255;
-	s[3] = u >> 8;
-}
-
-/**
- * \brief Pack the 32-bit integer in big-endian order into a buffer.
- * \param[out] s The buffer to write to.
- * \param u The integer to pack.
- */
-static inline void ut32_pack_big(char s[4], ut32 u) {
-	s[3] = u & 255;
-	u >>= 8;
-	s[2] = u & 255;
-	u >>= 8;
-	s[1] = u & 255;
-	s[0] = u >> 8;
-}
-
-/**
- * \brief Unpack the buffer, in low-endian order, into a 32-bit integer.
- * \param[in] s The buffer to read from.
- * \param[out] u The result of unpacking \p s.
- */
-static inline void ut32_unpack(char s[4], ut32 *u) {
-	ut32 result = 0;
-	result = (ut8)s[3];
-	result <<= 8;
-	result += (ut8)s[2];
-	result <<= 8;
-	result += (ut8)s[1];
-	result <<= 8;
-	result += (ut8)s[0];
-	*u = result;
-}
-
 #ifdef __cplusplus
 }
 #endif

